ir/shared/src/main/scala/org/scalajs/ir/Trees.scala

```diff
@@ -1203,7 +1203,7 @@ object Trees {
       case TopLevelJSClassExportDef(_, name) => name
 
       case TopLevelMethodExportDef(_, JSMethodDef(_, propName, _, _, _)) =>
-        val StringLiteral(name) = propName
+        val StringLiteral(name) = propName: @unchecked
```

Comment:
Could you add a comment that the annotation is required for Scala 3, please? Otherwise we could remove it by accident in the future, not realizing what it's for.

Reply:
Why is this required? Because the pattern can fail? If yes, IIRC, there are plenty of other places in this codebase where we use this.

Reply:
Yes, but it's the only one in `ir/`, apparently. And only `ir/` is recompiled from source with Scala 3.